projects/canopy/src/lib/utils/dynamic-style.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+
+import { BreakpointValues } from '../shared/breakpoints.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * Dynamically add styles to a <style> tag in <head>
+ */
+export class DyanmicStyleService {
+  styleTag = null;
+  selectors = [];
+  mediaQueries = {};
+
+  private createStyleTag(): void {
+    this.styleTag = document.createElement('style');
+    this.styleTag.type = 'text/css';
+    document.getElementsByTagName('head')[0].appendChild(this.styleTag);
+  }
+
+  private addMediaQuery(breakpoint: BreakpointValues): void {
+    if (!this.styleTag) {
+      this.createStyleTag();
+    }
+    if (this.mediaQueries[breakpoint]) {
+      return;
+    }
+    this.mediaQueries[breakpoint] = [];
+    this.styleTag.innerHTML += `@media(min-width:${breakpoint}){}`;
+  }
+
+  addStyle(selector: string, style: string): void {
+    if (!this.styleTag) {
+      this.createStyleTag();
+    }
+    if (this.selectors.includes(selector)) {
+      return;
+    }
+    this.selectors.push(selector);
+    const styleString = `.${selector}{${style}!important}`;
+    this.styleTag.innerHTML += styleString;
+  }
+
+  addStyleToMediaQuery(
+    selector: string,
+    style: string,
+    breakpoint: BreakpointValues,
+  ): void {
+    if (!this.styleTag) {
+      this.createStyleTag();
+    }
+    if (!this.mediaQueries[breakpoint]) {
```

Comment:
This kind of checking is interesting, should we do it here, or in the method itself, which can determine whether it should do something or return early. Not sure if there is a name for that approach, but doing it in the method almost feels safer? Interested in opinions on that one

Reply:
I think I know what you mean, what method are you referring to?